### PR TITLE
Suggestion #56

### PR DESCRIPTION
**Pull request by TomtheCoder2**
implemented Suggestion #56

### CODE REVIEW

**Comment by TomtheCoder2** [comment]
accidentally deleted the branch lol

File: core/src/mindustry/ui/CoreItemsDisplay.java [hunk 1]
```diff
@@ -6,15 +6,26 @@
...
+import mindustry.world.modules.*;
+
+import java.util.*;
 
 import static mindustry.Vars.*;
 
 public class CoreItemsDisplay extends Table{
     private final ObjectSet<Item> usedItems = new ObjectSet<>();
+    private final int maxAverage = 10;
+    private final HashMap<Item, Float> lastItemsAveraged = new HashMap<>();
```
Comment (buthed010203) [inline core/src/mindustry/ui/CoreItemsDisplay.java:18]: Use Arc's ObjectFloatMap instead.

File: core/src/mindustry/ui/CoreItemsDisplay.java [hunk 2]
```diff
@@ -6,15 +6,26 @@
...
+    private final ItemModule lastItems = new ItemModule();
     private CoreBuild core;
 
     public CoreItemsDisplay(){
         rebuild();
+        // add all items to the lists, else NullPointerException
```
Comment (buthed010203) [inline core/src/mindustry/ui/CoreItemsDisplay.java:24]: ObjectFloatMap get has a default for when its not found

**Review by buthed010203 — CHANGES_REQUESTED** [review]

**Comment by BalaM314** [comment]
Closing as I'm pretty sure this feature has been implemented

**Pull request closed without merging**